### PR TITLE
Management cleanup

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary

Fixes extant concerns from #313.

Puts all code that is specific to the current facility to be attached to the facility user resource.

In the future, this might be better put onto a facility Resource instead.

Cleans up code related to model creation and data setting, as we were previously calling some things in different ways, this ensures consistent and common code paths for the same operations.

### CODE REVIEW

File: kolibri/core/assets/src/apiResources/facilityUser.js [hunk 1]
```diff
@@ -1,9 +1,29 @@
...
+const logging = require('logging').getLogger(__filename);
 
 class FacilityUserResource extends Resource {
   static resourceName() {
     return 'facilityuser';
   }
+
+  getCurrentFacility() {
+    const promise = new Promise((resolve, reject) => {
+      this.resource.client({ path: this.currentFacilityUrl() }).then((response) => {
```
Comment (ralphiee22) [inline kolibri/core/assets/src/apiResources/facilityUser.js]: Isn't `this.resource` undefined in this context?

File: kolibri/core/assets/src/api_resource.js [hunk 2]
```diff
@@ -41,19 +41,13 @@ class Model {
   constructor(data, resource) {
     // Assign any data to the attributes property of the Model.
     this.attributes = {};
-    Object.assign(this.attributes, data);
+    this.set(data);
```
Comment (indirectlylit) [inline kolibri/core/assets/src/api_resource.js]: This line needs to be swapped with the one below it, because `this.set` references `this.resource`

Comment (rtibbles) [inline kolibri/core/assets/src/api_resource.js]: Excellent point!

**Comment by indirectlylit** [comment]
Looks like this merge caused the breakage we're currently seeing in master:

image

**Comment by ralphiee22** [comment]
Sorry, would this have been caught by the tests?

**Comment by indirectlylit** [comment]
Looks like before this change, calling `Resources.getModel(id).fetch()` with the `root_pk` returned an object that included all the keys:

* `pk`
* `title`
* `description`
* `ancestors`

now it just returns `pk`

**Comment by indirectlylit** [comment]
Also - note that the error occurs inside the Promise callback of `showExploreTopic`, but is being swallowed by the `catch` clause. If you comment out the `catch` you'll get a better error.

**Comment by indirectlylit** [comment]
> Sorry, would this have been caught by the tests?

We don't have much front-end testing at this point, so in general we need to manually screen and sanity-check every merge into master. No worries - lesson learned! :)

File: kolibri/core/assets/src/api_resource.js [hunk 3]
```diff
@@ -425,11 +410,10 @@ class Resource {
    * @param {String} id - The primary key of the Model instance.
    * @returns {Model} - Returns a Model instance.
    */
-  getModel(id, data = {}) {
```
Comment (ralphiee22) [inline kolibri/core/assets/src/api_resource.js:428]: @indirectlylit the problem happens with this new definition of getModel. Change it to the original and it works. Currently investigating..

**Pull request merged**